gradle/publishing.gradle

```diff
@@ -58,9 +58,41 @@ dependencies {
     mavenDeployer "org.apache.maven.wagon:wagon-ftp:2.8"
 }
 
-afterEvaluate {
-    assert (!projectPomName.isEmpty() && !projectPomDescription.isEmpty())
-    pom {
+uploadArchives { task ->
```

Comment:
Where do these parameters(username, pass, url)  get passed through now?  I see you trimmed down the the hard coded urls.

Reply:
I didn't change the way the parameters are passed. The mavenRepositoryUrl, mavenUsername and mavenUserPassword parameters are assumed to be passed by gradle properties, just like before. To explain the change that you have commented on:
Instead of adding the uploadArchives task in afterEvaluate, I moved the repository config to the doFirst block of uploadArchives task. I also changed the way that the existence of mavenRepositoryUrl, mavenUsername and mavenUserPassword parameters are checked. These parameters are only required when the uploadArchives task is part of the Gradle task graph and will be executed during the build. The [same approach was used in Gradle core build](https://github.com/gradle/gradle/blob/80be49b/gradle/publish.gradle#L47-L73).